### PR TITLE
Formatting with yapf

### PR DESCRIPTION
**Pull request by gautierronan**
Moving the discussion from https://github.com/PierreGuilmin/torchqdynamics/pull/6 to this PR. I played around with the formatters a little bit, and I quite like the yapf formatting with facebook style and a few other specific arguments. Mostly, it avoids wrapping long function definitions on many lines (which I find quite unreadable). Would this formatting be fine with you guys?

Sorry for being annoying on this haha. I feel like it's quite important to have as readable code as possible, since contrary to popular libraries, we won't have extensive documentation (for now at least).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -18,13 +18,17 @@ build-backend = "setuptools.build_meta"
...
-target-version = ["py38"]
-skip-string-normalization = true
+[tool.yapf]
+based_on_style = "facebook"
+column_limit = 88
+indent_width = 4
+SPLIT_BEFORE_NAMED_ASSIGNS = false
+DEDENT_CLOSING_BRACKETS = true
+SPLIT_ARGUMENTS_WHEN_COMMA_TERMINATED = false
+INDENT_DICTIONARY_VALUE = true
```
Comment (PierreGuilmin) [inline pyproject.toml:28]: I think these three guys are by default in the `"facebook"` style, so unnecessary to specify.

File: pyproject.toml [hunk 2]
```diff
@@ -18,13 +18,17 @@ build-backend = "setuptools.build_meta"
...
 
 [tool.isort]
 profile = "black"
-line_length = 80  # to be consistent with black
+line_length = 88  # to be consistent with yapf
 skip = ["__init__.py"]
+
```
Comment (PierreGuilmin) [inline pyproject.toml:34]: Could we remove the extra line?

File: pyproject.toml [hunk 3]
```diff
@@ -18,13 +18,17 @@ build-backend = "setuptools.build_meta"
...
-[tool.black]
-line-length = 80
-preview = true
-target-version = ["py38"]
-skip-string-normalization = true
+[tool.yapf]
+based_on_style = "facebook"
+column_limit = 88
+indent_width = 4
+SPLIT_BEFORE_NAMED_ASSIGNS = false
```
Comment (PierreGuilmin) [inline pyproject.toml:25]: Could we pick a uniform capitalization style?

File: pyproject.toml (hunk 2 above)
Comment (PierreGuilmin) [reply pyproject.toml:34]: You can tune your IDE so it does all this little things for you, for example in VSCode user configuration
```json
"files.trimTrailingWhitespace": true,
"files.insertFinalNewline": true,
"files.trimFinalNewlines": true,
"files.eol": "\n",
```

**Review by PierreGuilmin — CHANGES_REQUESTED** [review]
I'm fine with any formatter as long as we have one, I've been wanting to try yapf for a while, so I'm happy to go with it! 😊 Note that you also need to update `requirements-dev.txt`, `.pre-commit-config.yaml` and the `README.md`. Also, now would be a good time to reformat all the files and commit the modifications, as we don't have a lot of code and not too many PRs relying on existing code.

File: pyproject.toml (hunk 1 above)
Comment (gautierronan) [reply pyproject.toml:28]: The top and bottom ones yes, but not the middle one. Changing accordingly.

File: pyproject.toml (hunk 2 above)
Comment (gautierronan) [reply pyproject.toml:34]: Added that in my VSCode settings, thanks !

File: pyproject.toml (hunk 1 above)
Comment (PierreGuilmin) [reply pyproject.toml]: The middle one is in PEP8 style from which Facebook inherits.

**Comment by gautierronan** [comment]
Should be good now. I had to add a `setup.cfg` file to work around https://github.com/google/yapf/issues/936.

File: README.md [hunk 4]
```diff
@@ -22,17 +22,17 @@ conda develop /path/to/torchqdynamics
...
 ```shell
-pip install isort black
+pip install isort yapf
 ```
 
 Run the following before each commit:
 ```shell
 isort torchqdynamics
-black torchqdynamics
+yapf --style={based_on_style: facebook, column_limit: 88, split_before_named_assigns: False} torchqdynamics
```
Comment (PierreGuilmin) [inline README.md:31]: Weird, yapf should pick the `pyproject.toml` settings without needing to specify all of them. Maybe it's due to

> Should be good now. I had to add a setup.cfg file to work around https://github.com/google/yapf/issues/936.

I think a better fix is just to add `toml` as a dependency.

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -1,18 +1,32 @@
 files: torchqdynamics
 repos:
+  - repo: https://github.com/google/yapf
+    rev: v0.32.0
+    hooks:
+      - id: yapf
+        name: yapf
+        language: python
+        entry: yapf
+        args: [-i, -vv]
```
Comment (PierreGuilmin) [inline .pre-commit-config.yaml:10]: I think we can remove `-vv`, it will get very noisy when having more files.

**Review by PierreGuilmin — CHANGES_REQUESTED** [review]
Alright, almost done, nice little addons in the pre-commit config 🥓

**Comment by PierreGuilmin** [comment]
I will add some CI soon now with GitHub Actions now that we have so many cool checks

File: README.md (hunk 4 above)
Comment (gautierronan) [reply README.md:31]: Maybe passing the style here is useless indeed. I was not sure, so I did it anyway :D

Comment (PierreGuilmin) [reply README.md]: I think we need `yapf -i -r torchqdynamics`.

Comment (gautierronan) [reply README.md]: Ok I'm not quite sure about all this tbh. Can I let you make the proper changes on the pre-commit stuff?

**Review by PierreGuilmin — APPROVED** [review]

**Pull request merged**